### PR TITLE
Add "IP Proxy Handling" subsection

### LINKED ISSUE #100: Slightly ambiguous about how to process a request with a bad path
I'll just use a single example here because I think it makes it easier to express my point but the problems might be wider 

The URI template may include a `ipproto` variable, which is later defined in https://www.ietf.org/archive/id/draft-ietf-masque-connect-ip-05.html#section-4.5 as `ipproto = 1*3DIGIT / "*"`. The value can only be between 0 through 255, I presume this is a decimal value.

What is expected to happen if I send some bad values like:

* `:path = /proxy?ipproto=-1`
* `:path = /proxy?ipproto=abc`
* `:path = /proxy?ipproto=256`
* `:path = /proxy?ipproto=1234`

I presume that the recipient would reject the request with a bad path, by virtue of applying the requirement in section 4 that :path SHALL NOT be empty *and* following the expansion requirements in Section 3 *and* following those requirements on to Section 4.5. That's a lot of hops to get things right. Is there any easier way to present these?

### PR DESCRIPTION
**Pull request by DavidSchinazi**
Fixes #100

### CODE REVIEW

File: draft-ietf-masque-connect-ip.md [hunk 1]
```diff
@@ -183,34 +183,58 @@ Protocol (see {{Section 3.2 of HTTP-DGRAM}}) with HTTP Datagrams in the format
...
+
+Upon receiving an IP proxying request:
+
+ * if the recipient is configured to use another HTTP proxy, it will act as an
+   intermediary by forwarding the request to another HTTP server. Note that such
+   intermediaries may need to re-encode the request if they forward it using a
+   version of HTTP that is different from the one used to receive it, as the
+   request encoding differs by version (see below).
+
+* otherwise, the recipient will act as an IP proxy. It extracts the optional
```
Comment (achernya) [inline draft-ietf-masque-connect-ip.md:206]: markdown issue, missing space before *?

**Review by achernya — APPROVED** [review]

File: draft-ietf-masque-connect-ip.md (hunk 1 above)
Comment (DavidSchinazi) [reply draft-ietf-masque-connect-ip.md:206]:
```suggestion
 * otherwise, the recipient will act as an IP proxy. It extracts the optional
```

**Review by gloinul — APPROVED** [review]
Appear to look good.

File: draft-ietf-masque-connect-ip.md [hunk 2]
```diff
@@ -183,34 +183,58 @@ Protocol (see {{Section 3.2 of HTTP-DGRAM}}) with HTTP Datagrams in the format
...
+
+IP proxies MUST validate whether the decoded "target" and "ipproto" variables
+meet the requirements in {{scope}}. If they do not, the IP proxy MUST treat
+the request as malformed; see {{Section 8.1.1 of H2}} and
+{{Section 4.1.2 of H3}}. If the "target" variable is a DNS name, the IP proxy
+MUST perform DNS resolution before replying to the HTTP request. If errors
+occur during this process, the IP proxy MUST reject the request and SHOULD
+send details using an appropriate Proxy-Status header field
+{{!PROXY-STATUS=RFC9209}}. For example, if DNS resolution returns an error,
+the proxy can use the dns_error Proxy Error Type from
```
Comment (tfpauly) [inline draft-ietf-masque-connect-ip.md:219]:
```suggestion
the proxy can use the `dns_error` Proxy Error Type from
```

**Review by tfpauly — APPROVED** [review]

**Pull request merged**